### PR TITLE
added a StringToSuite method in suites for easier retrieval of suites

### PR DESCRIPTION
**Pull request by ineiti**
For easy encapsulation it is useful to not only have the `.String()`-method, but to also be able to do the reverse.

### CODE REVIEW

**Comment by ineiti** [comment]
Also added correct Unmarshalling of the 'infinite'-point in P256 and thus enabling that suite again.

File: suites/list.go [hunk 1]
```diff
@@ -3,24 +3,38 @@
...
 type Suites map[string]abstract.Suite
 
-func (s Suites) add(suite abstract.Suite) {
-	s[suite.String()] = suite
-}
-
+// Returns a map of all suites
 func All() Suites {
 	s := make(Suites)
+	// The P256-nist implementation doesn't work for the moment
```
Comment (Daeinar) [inline suites/list.go]: @ineiti: According to your latest note, P256 does work again and thus the above comment is redundant?

File: nist/curve.go [hunk 2]
```diff
@@ -179,10 +179,20 @@ func (p *curvePoint) MarshalBinary() ([]byte, error) {
 }
 
 func (p *curvePoint) UnmarshalBinary(buf []byte) error {
-	p.x, p.y = elliptic.Unmarshal(p.c, buf)
-	if p.x == nil || !p.Valid() {
-		return errors.New("invalid elliptic curve point")
+	// Check whether all bytes after first one are 0, so we
+	// just return the initial point.
```
Comment (Daeinar) [inline nist/curve.go]: I don't think that the current code below leads to a timing attack (which might happen with byte-by-byte comparisons where secret values are involved) but why take a risk? Maybe do something like the following?
```
var c byte = 0
for _, b := range buf[1:] {
		c |= b
}
if c != 0 {
	p.x, p.y = elliptic.Unmarshal(p.c, buf)
	if p.x == nil || !p.Valid() {
		return errors.New("invalid elliptic curve point")
	}
	return nil
}
...
```
I am probably just nitpicking...

Btw. are there somewhere utility functions in dedis/crypto which do constant time comparisons, like those on https://cryptocoding.net/index.php/Coding_rules#Compare_secret_strings_in_constant_time ? Might come in handy at various points.

**Comment by Daeinar** [comment]
Except for two small things, I think the changes are ok.

**Pull request merged**